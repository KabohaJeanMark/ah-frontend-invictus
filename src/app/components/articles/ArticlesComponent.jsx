import React from 'react';
import { Link } from "react-router-dom";
import '../../../assets/styles/articles.scss';

const ArticlesComponent = (props) => {
  const articlesItems = props.state.articles.articles.map(article => (
    <Link key={article.id} className="card-article-link" to={`/articles/${article.id}`}>
      <div className="card-article" style={{ width: 21.85 + 'rem' }}>
        <img className="card-image" alt="" src="https://picsum.photos/600/350" />
        <div className="card-body">
          <div><p className="card-publish-date">Published On: {article.created_at.substring(0, 10)}</p></div>
          <h4 className="card-title">{article.title}</h4><p className="card-text">{article.description}</p>
          <div className="card-body-footer">
            <p className="card-footer-text">@{article.author}</p>
            <p className="card-footer-text">{article.read_time}</p></div>
          <div className="card-body-likes-dislikes">
            <p className="card-body-likes-text">Likes: {' '}{article.likes_count}</p>
            <p className="card-body-dislikes-text">Dislikes: {' '}{article.dislikes_count}</p></div>
          <div className="all-tags"><hr />
            <ul>{article.tagList.map(tag => <li key={tag} className="tag">{tag}</li>)}</ul></div>
        </div></div>
    </Link>));
  return (
    <div>
      <div className="container">
        <h2 className="articles-heading">Recent Articles</h2><hr />
        <div className="row">{articlesItems}</div>
      </div>
    </div>
  );
};

export default ArticlesComponent;
