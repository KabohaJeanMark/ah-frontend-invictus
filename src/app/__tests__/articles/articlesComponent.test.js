import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ArticlesComponent from '../../components/articles/ArticlesComponent';

const props = {
  state: {
    articles: {
      articles: [{
        id: 1,
        created_at: 'Test',
        title: 'Test',
        description: 'Test',
        author: 'Test',
        tagList:["edna"],
        read_time: 'Test'
      }]
    }
  }
};

describe('articles component', () => {
  let wrapper = shallow(<ArticlesComponent {...props} />);

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
