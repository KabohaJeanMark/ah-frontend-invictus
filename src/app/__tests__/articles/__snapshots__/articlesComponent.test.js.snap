// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articles component matches snapshot 1`] = `
<div>
  <div
    className="container"
  >
    <h2
      className="articles-heading"
    >
      Recent Articles
    </h2>
    <hr />
    <div
      className="row"
    >
      <Link
        className="card-article-link"
        key="1"
        to="/articles/1"
      >
        <div
          className="card-article"
          style={
            Object {
              "width": "21.85rem",
            }
          }
        >
          <img
            alt=""
            className="card-image"
            src="https://picsum.photos/600/350"
          />
          <div
            className="card-body"
          >
            <div>
              <p
                className="card-publish-date"
              >
                Published On: 
                Test
              </p>
            </div>
            <h4
              className="card-title"
            >
              Test
            </h4>
            <p
              className="card-text"
            >
              Test
            </p>
            <div
              className="card-body-footer"
            >
              <p
                className="card-footer-text"
              >
                @
                Test
              </p>
              <p
                className="card-footer-text"
              >
                Test
              </p>
            </div>
            <div
              className="card-body-likes-dislikes"
            >
              <p
                className="card-body-likes-text"
              >
                Likes: 
                 
              </p>
              <p
                className="card-body-dislikes-text"
              >
                Dislikes: 
                 
              </p>
            </div>
            <div
              className="all-tags"
            >
              <hr />
              <ul>
                <li
                  className="tag"
                  key="edna"
                >
                  edna
                </li>
              </ul>
            </div>
          </div>
        </div>
      </Link>
    </div>
  </div>
</div>
`;
